### PR TITLE
move HandlePodCleanups function to right file

### PR DESCRIPTION
**Pull request by gongguan**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Kubelet's HandlePodXxx functions used in syncLoopIteration are all written in file kubelet.go except 
for HandlePodCleanups.
HandlePodCleanups function is posited at file kubelet_pods.go alone. I think it's more reasonable to move it to kubelet.go. at the same time, it's test functions are written in kubelet_test.go rather than kubelet_pod_test.go
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by gongguan** [comment]
```release-note
move HandlePodCleanups function to right position
```

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -2151,6 +2151,96 @@ func (kl *Kubelet) HandlePodSyncs(pods []*v1.Pod) {
 	}
 }
 
+// HandlePodCleanups performs a series of cleanup work, including terminating
```
Comment (mattjmcnaughton) [inline pkg/kubelet/kubelet.go]: Confirming that the test for this is already in `kubelet_test`?

**Review by mattjmcnaughton — COMMENTED** [review]
/ok-to-test

Provided the test suites pass, this makes sense to me in the short term. Mind pinging me when it passes and I will mark lgtm?

 In the longer term, I think `kubelet.go` is growing fairly large, so we may want to look to break it up. But that could be a later project, and it makes sense to aim for consistency for now.

Thanks for your pr :)

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (gongguan) [reply pkg/kubelet/kubelet.go]: yes

**Comment by gongguan** [comment]
> /ok-to-test
> 
> Provided the test suites pass, this makes sense to me in the short term. Mind pinging me when it passes and I will mark lgtm?
> 
> In the longer term, I think `kubelet.go` is growing fairly large, so we may want to look to break it up. But that could be a later project, and it makes sense to aim for consistency for now.
> 
> Thanks for your pr :)

ping @mattjmcnaughton tests passed

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

**Comment by gongguan** [comment]
@derekwaynecarr PTAL

**Comment by tallclair** [comment]
Seems like the other direction makes more sense? HandlePods* -> kubelet_pods.go? Doesn't really matter though, both files are huge.

> In the longer term, I think kubelet.go is growing fairly large, so we may want to look to break it up.

Totally agree, but I think the benefits of splitting the same object across files (as we have today) is marginal at best. I'd like to see someone split up the actual kubelet struct, ideally into multiple packages. Much bigger change though...

**Comment by gongguan** [comment]
So it's better to leave HandlePods* at two files just like now?
> I'd like to see someone split up the actual kubelet struct, ideally into multiple packages. Much bigger change though...

It's a huge project to split up the kubelet struct. If it needs to be done now, I'd like to give a hand as a volunteer

**Comment by gongguan** [comment]
@thockin could you please take a look?

**Comment by tallclair** [comment]
Yeah, I was advocating to move the HandlePods* functions instead. But again, I think just shuffling these functions between files isn't really addressing the root problem.

/area code-organization

**Comment by gongguan** [comment]
> Yeah, I was advocating to move the HandlePods* functions instead. But again, I think just shuffling these functions between files isn't really addressing the root problem.

In my perspective, moving HandlePod* functions together makes code consistency now. Splitting kubelet struct seems a little radical. @tallclair if it is welcome to implement, I'm glad to have a try.

**Comment by gongguan** [comment]
I noticed that Server(kubelet/server/server.go) used kubelet struct as implement of it's HostInterface, so override functions are all written at parent directory(kubelet_pods.go). It looks not good.
Can we split some kubelet member only used by server to a HostInterface implement  and move implement functions of HostInterface from kubelet file to Server file?

**Comment by derekwaynecarr** [comment]
I would prefer we write out our current structure before moving stuff around too broadly.  I know this has long been a discussion that @vishh and I had awhile back, but I am not sure I see a benefit in moving things around too broadly without a clear plan, even on a per package basis.

**Comment by gongguan** [comment]
> I am not sure I see a benefit in moving things around too broadly without a clear plan, even on a per package basis.

I'd like to ask a permission to split up kubelet struct. No function benefit, moving implement function from kubelet root package to sub package can light the load of kubelet.go and kubelet_pods.go file.

> I would prefer we write out our current structure before moving stuff around too broadly.

If the split makes sense, I can make a plan to write out which staff will be moved.

**Comment by tallclair** [comment]
If you're interested in tackling the broader Kubelet monolith problem, great! But you should probably start with a KEP.

I would start by documenting the current state. How are the kubelet components currently organized, and what are the dependencies between them? (there are currently a bunch of circular dependencies hidden through interfaces :nauseated_face: )

Then outline a plan for components that should be extracted from the kubelet, ideally with an updated dependency graph that is somewhat unraveled.

**Comment by gongguan** [comment]
> If you're interested in tackling the broader Kubelet monolith problem, great! But you should probably start with a KEP.
> 
> I would start by documenting the current state. How are the kubelet components currently organized, and what are the dependencies between them? (there are currently a bunch of circular dependencies hidden through interfaces 🤢 )
> 
> Then outline a plan for components that should be extracted from the kubelet, ideally with an updated dependency graph that is somewhat unraveled.

Sounds great! I'm glad to follow it and do something within my capabilities, thanks.

**Pull request closed without merging**